zingolib/src/wallet/transaction_context.rs

```diff
@@ -124,12 +107,14 @@ pub mod decrypt_transaction {
                             .recipient_address()
                             .map(|raw_taddr| address_from_pubkeyhash(&self.config, raw_taddr))
                         {
-                            outgoing_metadatas.push(OutgoingTxData {
-                                recipient_address: taddr,
-                                value: u64::from(vout.value),
-                                memo: Memo::Empty,
-                                recipient_ua: None,
-                            });
+                            if !taddrs_set.contains(&taddr) {
```

Comment:
yeah this single line looks like the bug fix we needed. nice

Reply:
https://github.com/zingolabs/zingolib/pull/1168/files#diff-c059f59e2dd106815b6b903751dd212059da1c2e8b7a0cec71db6fbcc8a48b27R536-R541
as you can see in the comment we need to be careful messing around with the definition of outgoing_tx_data as we could land a serious bug and block release.